### ISSUE TITLE
Unable to handle errors when loading the Google JavaScript Library

### ISSUE THREAD

**Issue by pjnicolas** [devbaji/vue3-google-login#74]
A few users are encountering a "Failed to load the Google 3P Authorization JavaScript Library" error on my app. This error is being captured by Sentry, and it appears that the library isn't handling or responding to the error appropriately.

Here's the relevant code snippet from this library:

```ts
export const loadGApi = () => new Promise<types.Google>((resolve, reject) => {
    // [...]

    script.addEventListener("error", () => {
      reject("Failed to load the Google 3P Authorization JavaScript Library.");
    });

    // [...]
});

export const googleSdkLoaded: types.GoogleSdkLoaded = (action) => {
    // [...]

    loadGApi().then((google) => {
      action(google);
    });
    // No catch block or promise handling here, leading to an unhandled rejection in the browser.

    // [...]
};
```

Is this behavior expected? If not, is there any temporary workaround to handle this error more gracefully?

**Comment by nielsvhb** [comment]
I am having the exact same problem. Sentry tells me this issue is quite common. Did you find anything so far?

**Comment by pjnicolas** [comment]
No, I just ignored the error on Sentry while I'm migrating to another tech stack.

**Comment by RobbeRls** [comment]
Created pull request https://github.com/devbaji/vue3-google-login/pull/77 for this issue this could allow you to pass an error property into the install to handle the exception yourself.

**Comment by devbaji** [comment]
Thanks @RobbeRls for the PR, merged it and your solution is now available in latest version.
